Added support to enable chassisd from the platform, which is necessary for SmartSwitch

rameshraghupathy opened the pull request:
Added support to enable chassisd form the platform which is required for SmartSwitch

#### Why I did it
SmartSwitch platform requires chassisd to run on fixed platform.  

#### How I did it
Added a flag "include_chassisd" which is enabled from the platform and will be enabled for SmartSwitch platform

#### How to verify iit
1. Added the flag in the code
2. Modified the platform SmartSwitch "../platform/pmon_daemon_control.json" file to enable it
3. Built an image and when loaded chassisd strated running on SmartSwitch image

#### Which release branch to backport (provide reason below if selected)
202311

#### Tested branch (Please provide the tested image version)
202311

#### Description for the changelog
{% if include_chassisd or not skip_chassisd and IS_MODULAR_CHASSIS == 1 %}

#### Link to config_db schema for YANG module changes
This PMON HLD like may be useful https://github.com/sonic-net/SONiC/blob/56d97c6e72884a3108618b191dc4570634304eaa/doc/smart-switch/pmon/smartswitch-pmon.md

#### A picture of a cute animal (not mandatory but encouraged)

vvolam commented:
Summary seems to be having a typo? Is it:
"Added support to enable chassisd from the platform, which is necessary for SmartSwitch"?

rameshraghupathy commented:
> Summary seems to be having a typo? Is it: "Added support to enable chassisd from the platform, which is necessary for SmartSwitch"?

Done

oleksandrivantsiv commented on dockers/docker-fpm-frr/base_image_files/TSA:
```diff
@@ -6,7 +6,16 @@ if [ "$EUID" -ne 0 ] ; then
 fi
 
-if [ -f /etc/sonic/chassisdb.conf ]; then
+PLATFORM="$(sonic-cfggen -d -v DEVICE_METADATA.localhost.platform)"
```
MSFT proposed a way to check if the system is a Smart Switch by accessing the `platform.json` file. The presence of the `DPUS` table means that the system is a Smart Switch NPU. The presence of the `DPU` table means that the system is a Smart Switch DPU: [Smart Switch IP Address Assignment](https://github.com/sonic-net/SONiC/blob/master/doc/smart-switch/ip-address-assigment/smart-switch-ip-address-assignment.md#device-data-and-platform).

This approach is used in various places by the Smart Switch Database service. I suggest using it here as well for consistency and to minimize the number of files that describe the system.

oleksandrivantsiv replied:
E.g. https://github.com/sonic-net/sonic-buildimage/pull/17562

rameshraghupathy replied:
Will follow that

rameshraghupathy replied:
Created https://github.com/sonic-net/sonic-buildimage/pull/19729  and addressed these comments  in #19729

oleksandrivantsiv reviewed (changes requested):
as commented

rameshraghupathy commented:
Created https://github.com/sonic-net/sonic-buildimage/pull/19729 and addressed these comments in https://github.com/sonic-net/sonic-buildimage/pull/19729.  #18676 18676 was created using the 202311 branch which is not where the smartswitch changes are applied.
